akka-diagnostics/src/main/scala/akka/diagnostics/ConfigChecker.scala

```diff
@@ -0,0 +1,1169 @@
+   *
+   * Potential configuration issues, if any, are printed to `System.out` and the JVM is exited with -1 status code.
+   *
+   * If no configuration issues are found the message "No configuration issues found" is printed to `System.out` and the
+   * JVM is exited with 0 status code.
+   *
+   * Use [#reportIssues] if you don't want to exit the JVM.
+   */
+  def main(args: Array[String]): Unit = {
+
+    val config = ConfigFactory
+      .parseString("akka.diagnostics.checker.fail-on-warning = on")
+      .withFallback(ConfigFactory.load())
+
+    Try(ActorSystem("ConfigChecker", config)) match {
+      case Success(_) =>
+        println("No configuration issues found")
+        System.exit(0)
+      case Failure(e) =>
+        println(e.getMessage)
+        Thread.sleep(2000) // give it a chance to flush log
+        System.exit(-1)
+    }
+  }
+
+  /**
+   * Validates the configuration of the given actor system. This is performed when the actor system is started.
+   */
+  def reportIssues(system: ExtendedActorSystem): Unit = {
+    import Internal._
+
+    def runChecks(): ValidationResults = {
+      val checker = new ConfigChecker(system)
+      val result = checker.check()
+      logWarnings(result)
+      result
+    }
+
+    def logWarnings(result: ValidationResults): Unit =
+      if (result.warnings.nonEmpty) {
+        val formatted = result.warnings.map(w => lightbendRecommendation(format(w)))
+        val log = LoggerFactory.getLogger(classOf[ConfigChecker].getName)
+        formatted.foreach(log.warn)
+      }
+
+    mode(system.settings.config) match {
+      case Disabled => // don't run checks
+      case LogWarnings =>
+        val asyncCheckAfter =
+          system.settings.config.getDuration("akka.diagnostics.checker.async-check-after", MILLISECONDS).millis
+        if (asyncCheckAfter > Duration.Zero)
+          system.scheduler.scheduleOnce(asyncCheckAfter)(runChecks())(system.dispatcher)
+        else
+          runChecks()
+
+      case FailOnWarnings =>
+        val result = runChecks()
+        if (result.warnings.nonEmpty)
+          throw new IllegalArgumentException(
+            lightbendRecommendation(result.warnings.map(format).mkString("\n* ", "\n* ", "\n")))
+    }
+  }
+
+  /**
+   * Formatted string representation of a warning.
+   */
+  def format(warning: ConfigWarning): String = {
+    val defaultsAsString = warning.defaultsAsString
+    s"${warning.message} Related config properties: [${warning.propertiesAsString}]. " +
+    (if (defaultsAsString == "") "" else s"Corresponding default values: [$defaultsAsString]. ") +
+    s"You may disable this check by adding [${warning.checkerKey}] to configuration string list " +
+    s"akka.diagnostics.checker.disabled-checks."
+  }
+
+  private def lightbendRecommendation(msg: String): String =
+    s"Lightbend recommendation: $msg Please use http://support.lightbend.com/ if you " +
```

Comment:
I think we should tone down the support aspect here. Could be 
```
s"Configuration recommendation: $msg"
```

and rename the `def` to `recommendation`